src/rule_definition_tools.jl

```diff
@@ -316,12 +320,20 @@ function _nondiff_frule_expr(primal_sig_parts, primal_invoke)
 end
 
 function _nondiff_rrule_expr(primal_sig_parts, primal_invoke)
-    num_primal_inputs = length(primal_sig_parts) - 1
+    has_vararg = _isvararg(primal_sig_parts[end])
+    if !has_vararg
+        num_primal_inputs = length(primal_sig_parts) - 1 # - primal
+        tup_expr = Expr(:tuple, ntuple(_->DoesNotExist(), num_primal_inputs)...)
+    else
+        num_primal_inputs = length(primal_sig_parts) - 2 # - primal and vararg
+        length_expr = :($(num_primal_inputs) + length($(_unconstrain(primal_sig_parts[end]))))
+        tup_expr = Expr(:call, :ntuple, Expr(:(->), :_, DoesNotExist()), length_expr)
+    end
```

Comment:
Do you think we should break this out into a function that just takes in the `primal_sig_parts` and returns a expression for the right number of `DoesNotExists()`?

Reply:
Thanks, was thinking about it and then forgot to implement in the end. Done now